### PR TITLE
Rely on Bolt data and not merchant database for referenc to parent quote

### PR DESCRIPTION
**Pull request by mccottry**
Our code base should move away from relying on client side data in any case that we can in order to avoid failures.  Furthermore, Bolt data is to be considered the source of truth for transactions.  For this reason, we continue with our pattern of using Bolt provided data for both quote and order lookups in preference to client side data stores.

### CODE REVIEW

File: app/code/community/Bolt/Boltpay/Model/Order.php [hunk 1]
```diff
@@ -57,13 +57,14 @@ public function createOrder($reference, $sessionQuoteId = null, $isPreAuthCreati
...
+            $immutableQuote->setParentId($transaction->order->cart->order_reference);
             Mage::app()->setCurrentStore($immutableQuote->getStore());
-            $parentQuote = $this->getQuoteById($immutableQuote->getParentQuoteId());
+            $parentQuote = $this->getQuoteById($transaction->order->cart->order_reference);
 
             if (!$parentQuote->getIsActive()) {
                 throw new Exception(
                     $this->boltHelper()->__("The parent quote %s for immutable quote %s is currently being processed or has been processed for order #%s. Check quote %s for details.",
-                        $parentQuote->getId(), $immutableQuote->getId(), $parentQuote->getReservedOrderId(), $parentQuote->getParentQuoteId() )
+                        $transaction->order->cart->order_reference, $immutableQuote->getId(), $parentQuote->getReservedOrderId(), $parentQuote->getParentQuoteId() )
```
Comment (matt-thomason) [inline app/code/community/Bolt/Boltpay/Model/Order.php]: Is this change necessary since you just redefined `$parentQuote` up there?

Comment (mccottry) [reply app/code/community/Bolt/Boltpay/Model/Order.php]: @matt-thomason You are right.   `$parentQuote->getId()` is useful information, but so is `$transaction->order->cart->order_reference`.  Let's iinclude them both.
